en/components/grids_templates/advanced_filtering.md

```diff
@@ -0,0 +1,457 @@
+    igx-chip: $custom-chip-schema,
+    igx-drop-down: $custom-drop-down-schema
+));
+
+$custom-grid: igx-grid-theme(
+    $palette: $grid-dark-palette,
+    $schema: $custom-light-schema
+);
+
+$custom-button: igx-button-theme(
+    $palette: $dark-palette,
+    $schema: $custom-light-schema
+);
+
+$custom-button-group: igx-button-group-theme(
+    $palette: $dark-palette,
+    $schema: $custom-light-schema
+);
+
+$custom-input-group: igx-input-group-theme(
+    $palette: $dark-palette,
+    $schema: $custom-light-schema
+);
+
+$custom-chip: igx-chip-theme(
+    $palette: $dark-palette,
+    $schema: $custom-light-schema
+);
+
+$custom-drop-down: igx-drop-down-theme(
+    $palette: $dark-palette,
+    $schema: $custom-light-schema
+);
+```
+
+Don't forget to include the themes in the same way as it was demonstrated above.
+
+#### Demo
+
+@@if (igxName === 'IgxGrid') {
+<div class="sample-container loading" style="height:530px">
+    <iframe id="grid-advanced-filtering-style-iframe" data-src='{environment:demosBaseUrl}/grid/grid-advanced-filtering-style' width="100%" height="100%" seamless="" frameBorder="0" class="lazyload no-theming"></iframe>
+</div>
+<br/>
+<div>
+<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="grid-advanced-filtering-style-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+}
+@@if (igxName === 'IgxTreeGrid') {
+<div class="sample-container loading" style="height:510px">
+    <iframe id="treegrid-advanced-filtering-style-iframe" data-src='{environment:demosBaseUrl}/tree-grid/treegrid-advanced-filtering-style' width="100%" height="100%" seamless="" frameBorder="0" class="lazyload no-theming"></iframe>
+</div>
+<br/>
+<div>
+<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="treegrid-advanced-filtering-style-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+}
+@@if (igxName === 'IgxHierarchicalGrid') {
+<div class="sample-container loading" style="height:630px">
+    <iframe id="hierarchical-grid-advanced-filtering-style-iframe" data-src='{environment:demosBaseUrl}/hierarchical-grid/hierarchical-grid-advanced-filtering-style' width="100%" height="100%" seamless="" frameBorder="0" class="lazyload no-theming"></iframe>
+</div>
+<br/>
+<div>
+<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="hierarchical-grid-advanced-filtering-style-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+}
+
+<div class="divider--half"></div>
+
+### API References
+<div class="divider--half"></div>
+
+* [IgxColumnComponent]({environment:angularApiUrl}/classes/igxcolumncomponent.html)
+* [@@igxNameComponent API]({environment:angularApiUrl}/classes/@@igTypeDoc.html)
+* [@@igxNameComponent Styles]({environment:sassApiUrl}/index.html#function-igx-grid-theme)
+
+### Additional Resources
```

Comment:
It is a good idea to add the other two filtering topics under additional resources as well.